pkg/controllers/ttl/controller.go

```diff
@@ -117,6 +117,31 @@ func (c *controller) deregisterEventHandlers() {
 	c.logger.V(3).Info("deregistered event handlers")
 }
 
+// Function to determine the deletion propagation policy
+func (c *controller) determinePropagationPolicy(metaObj metav1.Object, logger logr.Logger) *metav1.DeletionPropagation {
+	annotations := metaObj.GetAnnotations()
+	var policy *metav1.DeletionPropagation
+
+	if annotations != nil {
+		if annotationPolicy, ok := annotations["propagationPolicy"]; ok {
```

Comment:
You don't need `ok`, if the element is not found it will get you an empty string.